docs/docs/community/best-practices.md

```diff
@@ -0,0 +1,49 @@
+---
+sidebar_position: 5
+title: Best Practices
+---
+
+# Best Practices for Pull Requests and Contributions
+
+This guide provides essential best practices for submitting Pull Requests (PRs) and contributing effectively to the project. By adhering to these practices, contributors ensure a more efficient workflow, maintain code quality, and facilitate smooth collaboration within the community.
+
+## Pre-Review with AI
+
+Before submitting a Pull Request (PR), we strongly recommend conducting a pre-review using an AI tool, such as [Coderabbit](https://www.coderabbit.ai/) or [Sweep](https://www.coderabbit.ai/). This preliminary step helps identify potential issues and provides recommendations for improvement before human intervention. Addressing AI-generated feedback allows contributors to enhance the quality of their submission, ensuring that the subsequent human review is more focused, efficient, and substantive.
+
+## Checklist for Each New PR
+
+When creating a new PR, a checklist is automatically included through the PR template. Each item in this checklist must be addressed before the PR can be marked as "Ready for Review"; otherwise, the PR should remain in a draft (WIP) state. Here are some best practices we recommend:
+
+1. **Merge Latest Main**: Ensure your branch is up to date by merging the latest `main` branch (`git merge origin/main`).
+2. **Run Tests**: Execute all tests to verify that existing functionalities remain intact and unaffected by your changes.
```

Comment:
Can we add the command pnpm test. Also can we update the package.json to include running all tests not just tests under     "test": "pnpm --dir packages/core test"